pkg/labels/consul_aggregator.go

```diff
@@ -52,7 +52,7 @@ type consulAggregator struct {
 	path           string
 	kv             consulutil.ConsulLister
 	watchers       *selectorWatch
-	labeledCache   []Labeled // cached contents of the label subtree
+	labeledCache   *[]Labeled // cached contents of the label subtree
```

Comment:
So, elsewhere in this code I use `chan *[]Labeled`. This is primarily to signal to clients when there are 0 matches for a query instead of the watch being terminated for whatever reason. See http://play.golang.org/p/t8tlbBF9jO

On reflection, it's probably worth altering the chan to be `chan LabelWatchResult` and make that 0 vs nil distinction explicit.